lib/src/rules/implicit_reopen.dart

```diff
@@ -0,0 +1,231 @@
+// Copyright (c) 2023, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+
+import '../analyzer.dart';
+
+const _desc = r"Don't implicitly reopen classes or mixins";
+
+/// todo(pq): link out to (upcoming) dart.dev docs.
```

Comment:
The new class modifiers page creation is being tracked here https://github.com/dart-lang/site-www/issues/4497
I'll make a note in the todo there to make sure I come back here with the link once it's available (it will be `dart.dev/language/class-modifiers`)